src/core/search/search.h

```diff
@@ -55,13 +55,27 @@ class FieldIndices {
   absl::flat_hash_map<std::string, std::unique_ptr<BaseIndex>> indices_;
 };
 
+struct AlgorithmProfile {
+  struct ProfileEvent {
+    std::string descr;
+    size_t micros;  // time event took in microseconds
```

Comment:
I was using absl::Now() first and its not monotonic. I actually detected it just by checking the results visually 😆 So I switched to chrono's steady clock. 

I think the micros is fine. I don't see what the benefit is of using a scaringly long chrono type is if I just count the number of microseconds at the end

Reply:
What do you mean by non-monotonic? It represents a moment in time, regardless of timezones, and should never go back / forward in time
https://github.com/abseil/abseil-cpp/blob/fc44fa053cc91193d2bb60fb9356bcecb301242e/absl/time/time.h#L751-L752

Generally, I dislike having the units as a convention instead of part of the type, as it's error prone. Your call as always.

Reply:
absl::Now queries the system clock which is not monotonic, that's why I switched to chrono (and converting chrono timepoint to asbl duration isn't that nice as well)

https://github.com/abseil/abseil-cpp/issues/172#issuecomment-424432964

Reply:
it can go back in time since it uses a real clock. It's better using steady/monotonic clocks when measuring latency.
+1 to  @dranikpg preference to simplicity over standard. if to count microseconds I need to write:
`std::chrono::duration_cast<std::chrono::microseconds>(std::chrono::steady_clock::now() - start).count()` 
then maybe it's the standard committee that should do a better job.

Reply:
I didn't think of the case of the system changing time (I did think of daylight saving time changes, but that should be OK with absl::Now).
Anyway, I would use `std::chrono::microseconds` but I realize I'm in the minority. I saw _so many bugs_ caused by bad unit conversions, some [costlier than others](https://www.simscale.com/blog/nasa-mars-climate-orbiter-metric/). Sure, the type is verbose, but [there are std typedefs](https://en.cppreference.com/w/cpp/chrono/duration) to assist, and we can create one ourselves.
And.. in any case we need to write the long statement above (since `absl::Now()` is non monotonic), it's just a matter of storing a raw integer type vs a type-safe one
Having written that, @dranikpg you decide :)

Reply:
We're not sending spaceships to Mars, we're debugging profiling queries 😆

Its has nothing to do with dates. The system clock is non monotonic on a microsecond level and jumps around a little. Because its implemented by some low level device / caches values / corrects itself 🤷🏻‍♂️

Reply:
> We're not sending spaceships to Mars, we're debugging profiling queries 😆
Not yet. `SpaceFly` on the way to Mars :rocket: